core/app/models/spree/menu_item.rb

```diff
@@ -0,0 +1,84 @@
+module Spree
+  class MenuItem < Spree::Base
+    belongs_to :menu
+    belongs_to :linked_resource, polymorphic: true
+
+    acts_as_nested_set dependent: :destroy
+
+    before_save :reset_link_attributes, :build_path, :paremeterize_code
+
+    ITEM_TYPE = %w[Link Promotion Container]
+
+    LINKED_RESOURCE_TYPE = ['URL']
+    STATIC_RESOURCE_TYPE = ['Home Page']
+    DYNAMIC_RESOURCE_TYPE = ['Spree::Product', 'Spree::Taxon']
+
+    LINKED_RESOURCE_TYPE.unshift(*STATIC_RESOURCE_TYPE)
+    LINKED_RESOURCE_TYPE.push(*DYNAMIC_RESOURCE_TYPE)
+
+    validates :name, presence: true
+    validates :menu_id, presence: true, numericality: { only_integer: true }
```

Comment:
```suggestion
    validates :menu, presence: true
```

let's rely on association presence validation, also if someone would like to use UUIDs this would fail